From 5b4c9b5319d8c7a2dc1830d1bc9ffc81125644ba Mon Sep 17 00:00:00 2001
From: Petr Cvek <petrcvekcz@gmail.com>
Date: Thu, 19 Sep 2024 05:09:32 +0200
Subject: [PATCH 12/26] init PIT

technically only one initialization (in seabios) is needed
non IBM BIOS compatible payload doesn't have to need PIT
IBM BIOS compatible payload (esp. MSDOS) must have PIT initialized
---
 src/drivers/pc80/pc/i8254.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/src/drivers/pc80/pc/i8254.c b/src/drivers/pc80/pc/i8254.c
index e83f00e00f..30e9619730 100644
--- a/src/drivers/pc80/pc/i8254.c
+++ b/src/drivers/pc80/pc/i8254.c
@@ -2,10 +2,10 @@
 
 #include <arch/io.h>
 #include <commonlib/helpers.h>
-#include <cpu/x86/tsc.h>
 #include <delay.h>
 #include <pc80/i8254.h>
 
+
 /* Initialize i8254 timers */
 
 void setup_i8254(void)
@@ -16,13 +16,16 @@ void setup_i8254(void)
 	outb(0x00, TIMER0_PORT);
 
 	/* Timer 1 */
-	outb(TIMER1_SEL | LOBYTE_ACCESS | MODE3 | BINARY_COUNT,
+	outb(TIMER1_SEL | WORD_ACCESS | MODE2 | BINARY_COUNT,
 	     TIMER_MODE_PORT);
 	outb(0x12, TIMER1_PORT);
+	outb(0x0, TIMER1_PORT);
 }
 
 #define CLOCK_TICK_RATE	1193180U /* Underlying HZ */
 
+#if CONFIG(UNKNOWN_TSC_RATE)
+
 /* ------ Calibrate the TSC -------
  * Too much 64-bit arithmetic here to do this cleanly in C, and for
  * accuracy's sake we want to keep the overhead on the CTC speaker (channel 2)
@@ -94,7 +97,6 @@ bad_ctc:
 	return 0;
 }
 
-#if CONFIG(UNKNOWN_TSC_RATE)
 static u32 timer_tsc;
 
 unsigned long tsc_freq_mhz(void)
-- 
2.46.1

